ForkMe-Mobile/app/src/main/java/dlei/forkme/gui/activities/IntermediateActivity.java

```diff
@@ -0,0 +1,159 @@
+package dlei.forkme.gui.activities;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.content.pm.ActivityInfo;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.util.Log;
+
+import java.io.IOException;
+import java.util.Locale;
+
+import dlei.forkme.R;
+import dlei.forkme.datastore.DatabaseHelper;
+import dlei.forkme.datastore.NoDataException;
+import dlei.forkme.datastore.TooMuchDataException;
+import dlei.forkme.endpoints.BaseUrls;
+import dlei.forkme.endpoints.GithubApi;
+import dlei.forkme.gui.activities.github.TrendingRepositoriesActivity;
+import dlei.forkme.helpers.NetworkAsyncCheck;
+import dlei.forkme.helpers.NetworkHelper;
+import dlei.forkme.model.User;
+import dlei.forkme.state.AppSettings;
+import okhttp3.Interceptor;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+// Retrofit tutorials:
+// http://www.vogella.com/tutorials/Retrofit/article.html
+// https://github.com/codepath/android_guides/wiki/Consuming-APIs-with-Retrofit
+// https://www.youtube.com/watch?v=R4XU8yPzSx0
+// http://square.github.io/retrofit/
+/**
+ * Due to restrictions in OAuth Library, need this activity to load in the user token into a
+ * static string which is better than shared preferences. Also ensures no events can happen before
+ * user details are retrieved from getAuthenticatedUser().
+ */
+public class IntermediateActivity extends AppCompatActivity {
+
+    private DatabaseHelper mDbHelper;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_intermediate);
+        setRequestedOrientation (ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
+
+
+        // Safer to have OAuth token as static variable as afaik shared preferences is just XML.
+        SharedPreferences sharedPreferences = getSharedPreferences("github_prefs", 0);
+        AppSettings.setOAuthToken(sharedPreferences.getString("oauth_token", null));
+
+        // Remove OAuth token from shared preferences.
+        SharedPreferences.Editor editor = sharedPreferences.edit();
+        editor.putString("github_prefs", "");
+        editor.apply();
+        editor.commit();
+
+        this.getAuthenticatedUser();
+    }
+
+
+    /**
+     * Get a GitHub user who logged in via OAuthToken.
+     */
+    public void getAuthenticatedUser() {
+        OkHttpClient.Builder okHttpBuilder = new OkHttpClient.Builder();
```

Comment:
You could probably put this network kind of stuff in a singleton and ideally in a Service, not an Activity.